media/CircleEditor/view-sidebar.js

```diff
@@ -1456,30 +1437,267 @@ sidebar.ArgumentView = class {
         return true;
     }
 
-    /**
-     * TODO Implement compareData()
-     * Compare and change transformed data against corrected data
-     */
+    compareData(bufferArr, originalArr, originalBuf, arrType) {
+        let dataLength = arrType === 4 ? bufferArr.length * 2 : bufferArr.length;
+        for (let i = 0; i < dataLength; i++) {
+            /* compare string formatted buffer data */
+            if (originalArr[i] !== bufferArr[i]) {
+                /* when data change detected, update buffer according to type */
+                var buffer = new ArrayBuffer(8);
+                const view = new DataView(buffer);
+                switch (arrType) {
+                    case 'float32':
+                        view.setFloat32(0, parseFloat(bufferArr[i]), true);
+                        for (let j = 0; j < 4; j++) {
+                            originalBuf[i * 4 + j] = view.getUint8(j);
+                        }
+                        break;
```

Comment:
We originally had the dataType array where the 4th index was 'int64', and since int64 takes twices as much space then other types and that is why `bufferArr.length` is multiplied by 2. We forgot to change that part once we removed the type array, sorry about that!

However, we still haven't figured out how we are going to handly data modification when current type is int64 or uint64, because the netron shows them as "high" and "low" format. Therefore, we will put all int64 and uint64 as TODO for now:(

Also, the we are currently comparing `originalArr[i]` and `bufferArr[i]` so that we can update only the ones that are modified. If we remove this clause, we will be updating every single buffer data. Is that okay?

Reply:
> Also, the we are currently comparing originalArr[i] and bufferArr[i] so that we can update only the ones that are modified. If we remove this clause, we will be updating every single buffer data. Is that okay?

Yes :)
I think that this is not that critical for the performance because loop count is invariant while update logic is not that expensive :D